### PR TITLE
Minor fixes in matrix.cpp

### PR DESCRIPTION
**Pull request by valeriyvan**
Fixes indentation.
Moves local vars declarations to places where they are used.
Uses NULL instead of 0 for pointers.

### CODE REVIEW

File: modules/core/src/matrix.cpp [hunk 1]
```diff
@@ -3426,13 +3422,14 @@ reduceC_( const Mat& srcmat, Mat& dstmat )
...
         else
         {
-            for( k = 0; k < cn; k++ )
+            for( int k = 0; k < cn; k++ )
             {
                 WT a0 = src[k], a1 = src[k+cn];
+                int i;
                 for( i = 2*cn; i <= size.width - 4*cn; i += 4*cn )
                 {
                     a0 = op(a0, (WT)src[i+k]);
```
Comment (terfendail) [inline modules/core/src/matrix.cpp]: Could you please indent line 3446 `dst[k] = (ST)a0;` as well?

**Comment by terfendail** [comment]
Could you please squash all commits into one?

File: modules/core/src/matrix.cpp (hunk 1 above)
Comment (valeriyvan) [inline modules/core/src/matrix.cpp]: done

**Comment by valeriyvan** [comment]
made new PR with squashed commits

**Pull request closed without merging**